### PR TITLE
[4.8 feature] vSphere deployment with thick storageclass

### PR DESCRIPTION
**Pull request by ebenahar**
https://issues.redhat.com/browse/KNIP-1610

According to https://bugzilla.redhat.com/show_bug.cgi?id=1932280#c10:

There is no "diskformat: thick" supported in OCP. There is only "thin", "zeroedthick" and "eagerzeroedthick" - see https://docs.openshift.com/container-platform/4.7/storage/dynamic-provisioning.html#vsphere-definition_dynamic-provisioning

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: conf/deployment/vsphere/thick_storage_class.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+# This config is suppose to work on most of DCs we have.
+DEPLOYMENT:
+  default_storage_class: 'thick'
```
Comment (prsurve) [inline conf/deployment/vsphere/thick_storage_class.yaml:4]: Can we make this as thick_sc_ocs_deploy true

File: conf/deployment/vsphere/thick_storage_class.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  storage_class_yaml: 'apiVersion: storage.k8s.io/v1
+                                kind: StorageClass
+                                metadata:
+                                  name: thick
+                                parameters:
+                                  datastore: vsanDatastore
+                                  diskformat: thick
+                                provisioner: kubernetes.io/vsphere-volume
+                                reclaimPolicy: Delete
+                                volumeBindingMode: Immediate'
```
Comment (prsurve) [inline conf/deployment/vsphere/thick_storage_class.yaml:14]: this can be moved to new YAML thick_sc.yaml

**Comment by ebenahar** [comment]
Marking as verified, even though the deployment with thick provisioning is blocked due to https://bugzilla.redhat.com/show_bug.cgi?id=1932280.

We can merge this PR

**Review by dahorak — APPROVED** [review]

**Comment by ebenahar** [comment]
Verified:
Cluster Configuration: conf/deployment/vsphere/zeroedthick_storage_class.yaml
PR Test Suite: tier1
PR Test Path: tests/
Additional Test Params:
OCP VERSION: 4.7
OCS VERSION: 4.7

https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/951/

**Review by AviadP — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**